### ISSUE TITLE
spreads-sdist.tar.gz

### ISSUE THREAD

**Issue by boredland** [DIYBookScanner/spreadpi#34]
Hi there,

I guess spreads-sdist.tar.gz is a precompiled version of spreads and depencies. How is it created?

Yours
BoredLand

**Comment by duerig** [comment]
Check out the spreads repository, enter the directory, then run:

python setup.py sdist

It will create a dist folder with a tarball that you can install with
pip and/or feed to the SpreadPi script.

On Thu, Jun 25, 2015, at 03:47 PM, Jonas Strassel wrote:

> Hi there,
> 
> I guess spreads-sdist.tar.gz is a precompiled version of spreads and
> depencies. How is it created?
> 
> Yours
> 
> BoredLand
> 
> — Reply to this email directly or view it on GitHub[1].

Links:
1. https://github.com/DIYBookScanner/spreadpi/issues/34

**Comment by boredland** [comment]
Thats nice. I guess this should be done in an armhf environment?

**Comment by duerig** [comment]
It doesn't matter what environment it is done in, as long as there are
the build tools that it needs. The build process is just Python
compilation and JS file stitching. There is nothing that gets compiled
down to machine code. I usually run it on my development Ubuntu VM
without doing anything special to try to make it seem like an arm
processor.

-D

On Thu, Jun 25, 2015, at 03:54 PM, Jonas Strassel wrote:

> Thats nice. I guess this should be done in an armhf environment?
> 
> — Reply to this email directly or view it on GitHub[1].

Links:
1. https://github.com/DIYBookScanner/spreadpi/issues/34#issuecomment-115410387

**Comment by boredland** [comment]
Awesome. Does the spreadpi-script work with only a symlink to spreads-sdist.tar.gz from the delivery/files folder?

**Comment by duerig** [comment]
There is a command line option you can use to point it at any tarball
location. I think it copies or symlinks the path you point it at to the
spreads-sdist.tar.gz location.

FYI, the present SpreadPi script has a number of issues that are not
ideal which is why I'm working on a replacement using Spindle. In the
longer term, it would be good to have a Spindle-like build system for
both the Pi and for the Android images you are working on. Among other
things, Spindle uses COW images so that you only need to do slow things
once (like install all the base packages). Which means you can re-run
later stages (compile and install a quick bugfix) without waiting all
day for the same version of all the dependencies to install first.

On Thu, Jun 25, 2015, at 04:00 PM, Jonas Strassel wrote:

> Awesome. Does the spreadpi-script work with only a symlink to spreads-
> sdist.tar.gz from the delivery/files folder?
> 
> — Reply to this email directly or view it on GitHub[1].

Links:
1. https://github.com/DIYBookScanner/spreadpi/issues/34#issuecomment-115412772

**Comment by boredland** [comment]
That'd be awesome. Testing the script is quite a torture right now. By the way: it works. The first built image will be online in some minutes:
http://ec2-52-28-157-49.eu-central-1.compute.amazonaws.com/

**Comment by markvdb** [comment]
You are aware of the images at http://buildbot.diybookscanner.eu? Just
making sure...

2015-06-26 0:09 GMT+02:00 Jonas Strassel notifications@github.com:

> That'd be awesome. Testing the script is quite a torture right now. By the
> way: it works. The first built image will be online in some minutes:
> http://ec2-52-28-157-49.eu-central-1.compute.amazonaws.com/
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/DIYBookScanner/spreadpi/issues/34#issuecomment-115415797
> .

## 

Mark Van den Borre
Hogestraat 16
3000 Leuven, België
+32 486 961726

**Comment by boredland** [comment]
Yes, using them all the time. Was just wondering what that part meant in
the build-script.

2015-06-26 1:15 GMT+02:00 Mark Van den Borre notifications@github.com:

**Comment by boredland** [comment]
Closing this myself as the reporter — I'm no longer pursuing it, and I'd rather not leave it sitting in your triage queue.

No action is needed from the maintainers, and this says nothing about the validity of the report: if anyone else is hitting the same thing, please reopen or file fresh.

**Comment by boredland** [comment]
Reopening — please disregard my previous comment.

That closure was a mistake on my side: it came from a bulk sweep of my own filed issues, not from a considered decision to withdraw this report. Sorry for the noise in your notifications.

The report stands as originally written.
